sydent/db/profiles.py

```diff
@@ -38,3 +38,22 @@ def addBatch(self, host, batchnum, batch):
             vals,
         )
         self.sydent.db.commit()
+
+    def getProfilesMatchingSearchTerm(self, search_term, limit):
+        cur = self.sydent.db.cursor()
+
+        sql = (
+            "SELECT user_id, display_name, avatar_url FROM profiles WHERE "
+            "user_id LIKE LOWER(?) OR "
+            "LOWER(display_name) LIKE LOWER(?) "
+            "LIMIT ?"
+        )
+
+        like_pat = '%' + search_term.replace('%', '%%') + '%'
```

Comment:
technically we should be escaping other like wildcard chars here, i guess - i.e. `_`.